src/core/execute.c

```diff
@@ -3189,6 +3189,24 @@ static int apply_mount_namespace(
                 }
         }
 
+        if (MANAGER_IS_SYSTEM(u->manager)) {
+                propagate_dir = path_join("/run/systemd/propagate/", u->id);
+                if (!propagate_dir)
+                        return -ENOMEM;
+                incoming_dir = strdup("/run/host/incoming");
+                if (!incoming_dir)
+                        return -ENOMEM;
+        } else if (context->protect_home == PROTECT_HOME_NO || context->protect_home == PROTECT_HOME_TMPFS) {
+                const char *suffix = strjoina("/systemd/propagate/", u->id);
```

Comment:
this is problematic. both left-hand and the right-hand side allocate on stack, traditionally C compilers had issues with that. I'd do that in two lines

Reply:
I will never remember all the rules about alloca&friends. Fixed, thanks